utils/get_data.py

```
@@ -1,27 +1,127 @@
+    """
+    Function to get the JSON with asset data from the Lizard API.
+
+    username is the username used for logging into Lizard.
+    password is the password used for logging into Lizard.
+    """
+    # Download all the assets of an asset type
+    max_amount = get_max_amount(username, password, asset_type)
+    payload = {"async": "true", "format": PAYLOAD_FORMAT,
+               "page_size": DOWNLOAD_LIMIT, "page": 1}
+    current_amount = 0
+    results = []
+    queue_time_interval = TASK_INTERVAL_MIN
+
+    while current_amount < max_amount:
+        # Create a task to download the JSON async
+        task_start_url = "{}{}/".format(BASE_URL, asset_type)
+        task_start_json = get_json(username, password, task_start_url, payload)
+
+        # Get the task JSON
+        task_payload = {"format": "json"}
+        task_url = "{}".format(task_start_json["url"])
+        task_json = get_json(username, password, task_url, task_payload)
+
+        # Check whether the task is done
+        task_status = str(task_json["task_status"])
+        while task_status == TASK_STATUS_PENDING:
+            # Update the task JSON
+            task_json = send_request(task_url, username, password)
+            task_status = str(task_json["task_status"])
+            time.sleep(queue_time_interval)
+            # Increase the task interval (in seconds) for bigger tasks to
+            # decrease the amount of requests done
+            if queue_time_interval < TASK_INTERVAL_MAX:
+                queue_time_interval += TASK_INTERVAL_INCREMENT
+        if task_status == TASK_STATUS_SUCCESS:
+            # Get the JSON with the data
+            result_url = task_json["result_url"]
+            result_json = send_request(result_url, username, password)
+            for result in result_json["results"]:
+                results.append(result)
+        elif task_status == TASK_STATUS_FAILURE:
+            dockwidget.set_all_status_bars_text("Task failure.")
```

I think we should throw an exception here, since failure is an exceptional situation (probably), and catch that exception elsewhere.

Moreover, this function becomes simpler without the ``dockwidget`` as argument.